### PR TITLE
OS Theme shining though fix 

### PR DESCRIPTION
**Pull request by daschuer**
This fixes https://bugs.launchpad.net/mixxx/+bug/1773709, and should help in 
https://github.com/mixxxdj/mixxx/pull/1687
and
https://github.com/mixxxdj/mixxx/pull/1677

### CODE REVIEW

**Comment by ronso0** [comment]
Thanks, it works!
2 1 1-skin-fixes-tango-preview-now bg-fix

**Comment by daschuer** [comment]
I have cleaned up the solution a bit.
Who can help with test and review?

**Comment by nopeppermint** [comment]
still working, no OS theme shining though, but I now get more frequently this behavior:
two orange play buttons...
preview

**Comment by ronso0** [comment]
> still working, no OS theme shining though, but I now get more frequently this behavior:
two orange play buttons...

I can see the previously activated preview button light up shortly when I hit another preview button.
But I noticed it before those changes.

**Comment by nopeppermint** [comment]
> button light up shortly

it's not shortly, it's persistent..

**Comment by nopeppermint** [comment]
and I do not have this on #1687

**Comment by nopeppermint** [comment]
>   > button light up shortly

> it's not shortly, it's persistent..

prober rebuild fixed everything ..

**Comment by ronso0** [comment]
@daschuer Before merging this, I think it's better to reset to https://github.com/mixxxdj/mixxx/pull/1691/commits/d3866763d73e7f30a80080b23445e551b211a2e9 to avoid conflicts in #1677

**Comment by nopeppermint** [comment]
BTW:
there is more OS Theme shining through if you edit in the library:
os_theme
but in the bpm folder it's already fixed
bpm_fixed

**Comment by ronso0** [comment]
> there is more OS Theme shining through if you edit in the library

I tried `QTableView::item:edit-focus` but that didn't work.

**Comment by daschuer** [comment]
OK, rebased, to get rid of merge conflicts. 
Does anyone likes to do a code review. 
I like to release 2.1 soon, and it would be a pity if this will not be included.

File: src/library/bpmdelegate.cpp [hunk 1]
```diff
@@ -75,6 +77,21 @@ void BPMDelegate::paint(QPainter* painter,const QStyleOptionViewItem &option,
     QStyleOptionViewItemV4 opt = option;
     initStyleOption(&opt, index);
 
+    // Set the palette appropriately based on whether the row is selected or
```
Comment (uklotzde) [inline src/library/bpmdelegate.cpp]: These lines are duplicated in TableItemDelegate::paint(). Can we move them into a protected utility function in the base class?

**Comment by uklotzde** [comment]
I'm not a UI expert, but the changes look fine so far. Code quality has been improved. Only a small amount of code duplication that could be avoided. I will test this now.

**Comment by uklotzde** [comment]
Here is a first screenshots from an **inofficial Qt5** build. Later I will switch back to the Qt4 build and test again.

When selecting a BPM cell the value becomes invisible:
bpm locked selected

This worked correctly before applying these changes.

**Comment by nopeppermint** [comment]
@uklotzde 
> When selecting a BPM cell the value becomes invisible:

 In my opinion you need as well those UI changes from #1677

**Comment by uklotzde** [comment]
On Qt4 I don't even notice any visual differences in Deere (all previous changes included). My only complaint is the small amount of duplicated code.

**Comment by daschuer** [comment]
Good catch, it was not only duplicated code It was redundant code. I have probably failed to remove it after moving the code.

**Comment by daschuer** [comment]
@uklotzde: ready for merge?

**Comment by uklotzde** [comment]
Windows build failures seem to be unrelated.

I did not notice the bugs in my personal setup and configuration, but I trust the other reviewers that the original issues have been fixed. The code improvements are some welcome added value.

LGTM.

**Pull request merged**